test/Microsoft.ML.TestFramework/TestCommandBase.cs

```diff
@@ -2028,5 +2028,15 @@ public void CommandTrainingBinaryFieldAwareFactorizationMachineWithValidationAnd
             Assert.True(outputPath.CheckEqualityNormalized());
             Done();
         }
+
+        [Fact]
+        public void DataTypes()
+        {
+            string idvPath = GetDataPath("datatypes.idv");
+            OutputPath textOutputPath = CreateOutputPath("datatypes.txt");
+            TestCore("savedata", idvPath, "loader=binary", "saver=text", textOutputPath.Arg("dout"));
+            Contracts.Check(textOutputPath.CheckEquality());
```

Comment:
>Contracts.Check [](start = 12, length = 15)

I think you just meant `Check`, not `Contracts.Check`. Also you'll want to include a sensible error message.

Reply:
Also I think `TestCore` should be running the comparison itself (is it not) so I don't think this line serves any purpose.

---
In reply to: [213156468](https://github.com/dotnet/machinelearning/pull/750#discussion_r213156468) [](ancestors = 213156468)